### PR TITLE
avoid restoring zeros in reporting

### PR DESCRIPTION
**Pull request by fbenke-pik**
This PR avoids using the memory-intense option restoreZeros in reportEmi and reportFE to read in data, mostly by introducing [new options when reading in data from gdxes](https://github.com/pik-piam/gdx/pull/8).

I tested with various gdxes from previous runs that reporting does not break and summation gaps did not widen through this.

### CODE REVIEW

**Comment by mellamoSimon** [comment]
great, thank you for this, Falk! which remind runs did you use to run your tests? we introduced some changes within remind that affect those variables and it would be great to have a comparison that includes those changes in the resulting `gdx`s and also be able to compare ceteris paribus, basically to make sure that the differences you see in the summation checks you mentioned [here](https://github.com/pik-piam/remind2/issues/549#issuecomment-1991294734) are not coming from this changes. Or maybe you checked already and I didn't get it? thank you!

**Comment by mellamoSimon** [comment]
> we introduced some changes within remind that affect those variables and it would be great to have a comparison that includes those changes

this is the latest one https://github.com/remindmodel/remind/pull/1599

also tagging @0UmfHxcvx5J7JoaOhFSs5mncnisTJJ6q and @fschreyer so they can stay on top of things :)

**Review by LaviniaBaumstark — APPROVED** [review]
thanks

**Comment by fbenke-pik** [comment]
I tested with these runs
```
  "SDP_MC-NPi-AMT_2024-03-01_22.13.42",
  "SSP2EU_PBS-NDC-AMT_2024-03-02_04.26.14",
  "SSP2EU_PBS-NPi-AMT_2024-03-01_22.14.22",
  "SSP2EU_PBS-PkBudg650-AMT_2024-03-02_04.26.25",
  "SSP2EU-EU21-PkBudg1050-AMT_2024-03-02_17.16.02",
  "SSP2EU-EU21-PkBudg500-AMT_2024-03-02_17.15.46",
  "SSP2EU-EU21-PkBudg650-AMT_2024-03-02_17.15.54",
  "SSP2EU-NDC-AMT_2024-03-02_06.05.42",
  "SSP2EU-NPi-AMT_2024-03-01_22.11.59",
  "SSP2EU-PkBudg1050-AMT_2024-03-02_06.06.08",
  "SSP2EU-PkBudg500-AMT_2024-03-02_06.05.52",
  "SSP2EU-PkBudg650-AMT_2024-03-02_06.06.01",
  "SSP5-NPi-AMT_2024-03-01_22.13.21",
  "SSP5-PkBudg650-AMT_2024-03-02_06.06.32"
 ```


I see changes in summations, but all are improvements and they are most likely due to some of your fixes.

**Comment by mellamoSimon** [comment]
maybe Michaja @0UmfHxcvx5J7JoaOhFSs5mncnisTJJ6q has managed to get some recent runs?

**Review by 0UmfHxcvx5J7JoaOhFSs5mncnisTJJ6q — CHANGES_REQUESTED** [review]
Honestly, I am at a loss as to why this works.
Simón said he required `restore_zeros = TRUE` during development [[↑]](https://github.com/pik-piam/remind2/issues/549#issuecomment-1980456161) and there is no material change in the code.
This
> mostly by introducing [new options when reading in data from gdxes](https://github.com/pik-piam/gdx/pull/8).

actually only affects `pm_IndstCO2Captured` (and does not work there, so it needs to be changed).

I guess `readGDX(restore_zeros = FALSE)` works for the variables because they have non-zero marginals across `ttot`, but if that was not the case earlier (see above) it might change again in the future.
`¯\_(ツ)_/¯`

File: R/reportEmi.R [hunk 1]
```diff
@@ -145,21 +145,19 @@ reportEmi <- function(gdx, output = NULL, regionSubsetList = NULL, t = c(seq(200
...
   }
-# CO2 captured per industry subsector
-  # note: this has to be read in with restore_zeros=T because sometimes it contains only non-zero values for "ETS" emiMkt
-  # and magclass will wrongly interpret this as a region
-  pm_IndstCO2Captured <- readGDX(gdx, "pm_IndstCO2Captured", restore_zeros = T,
-                                 react = 'silent')[,t,]
+
+  # CO2 captured per industry subsector
+  pm_IndstCO2Captured <- readGDX(gdx, "pm_IndstCO2Captured", regional = 1, spatial = 2, restore_zeros = FALSE, react = "silent")
+  pm_IndstCO2Captured <- pm_IndstCO2Captured[, intersect(getYears(pm_IndstCO2Captured), t), ]
```
[the remark this answers is not in the document]
Comment (fbenke-pik) [reply R/reportEmi.R:151]: Thanks for pointing that out! 

As `pm_IndstCO2Captured` is read in from gdx and then discarded right now, I will remove these lines. Whenever anyone is ready to bring it back, I can help with making sure that this won't crash (e.g. by introducing the helper to expand magclass objects discussed [here](https://github.com/pik-piam/remind2/issues/549))

**Comment by fbenke-pik** [comment]
> Honestly, I am at a loss as to why this works. Simón said he required `restore_zeros = TRUE` during development [[↑]](https://github.com/pik-piam/remind2/issues/549#issuecomment-1980456161) and there is no material change in the code. This
> 
> > mostly by introducing [new options when reading in data from gdxes](https://github.com/pik-piam/gdx/pull/8).
> 
> actually only affects `pm_IndstCO2Captured` (and does not work there, so it needs to be changed).
> 
> I guess `readGDX(restore_zeros = FALSE)` works for the variables because they have non-zero marginals across `ttot`, but if that was not the case earlier (see above) it might change again in the future. `¯\_(ツ)_/¯`

I introduced the `spatial` argument whenever sth. crashed when testing with some gdxes (usually, when the only value for emission market in the read in data was "ETS", which makes magclass think, it must be a regional column). 

If we cannot reproduce the things that made this crash in the past, I'd say we go ahead and merge. The moment at happens again, I will work on fixing it. It is just pretty difficult to fix sth. I cannot reproduce.

**Comment by 0UmfHxcvx5J7JoaOhFSs5mncnisTJJ6q** [comment]
> I introduced the `spatial` argument whenever sth. crashed when testing with some gdxes (usually, when the only value for emission market in the read in data was "ETS", which makes magclass think, it must be a regional column), so should be more than one problem.

Indeed, my eyes glazed over when looking at the diff.

**Review by 0UmfHxcvx5J7JoaOhFSs5mncnisTJJ6q — APPROVED** [review]

File: R/reportEmi.R (hunk 1 above)
[the remark this answers is not in the document]
Comment (0UmfHxcvx5J7JoaOhFSs5mncnisTJJ6q) [reply R/reportEmi.R]: Subject to #524.

**Pull request merged**